### PR TITLE
Add more unit test coverage

### PR DESCRIPTION
**Pull request by insacuri**
# What and why?

# How to test?

# Trello

### CODE REVIEW

File: test/controllers/test_party_controller.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            password_reset_counter=0,
+        )
+        row_2 = Row(
+            id="2",
+            party_uuid=respondent_2_id,
+            status="ACTIVE",
+            email_address="petergriffin@email.com",
+            pending_email_address="",
+            first_name="Peter",
+            last_name="Griffin",
```
Comment (LJBabbage) [inline test/controllers/test_party_controller.py:101]: Family survey?

File: test/controllers/test_case_controller.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        mock_engine.engine.execute().all.return_value = [row_1]
+        expected_output = [row_1]
+        test_output = case_controller.get_exercise_completion_stats(exercise_id)
+        self.assertEqual(getattr(expected_output[0], "Sample Size"), getattr(test_output[0], "Sample Size"))
+        self.assertEqual(getattr(expected_output[0], "Not Started"), getattr(test_output[0], "Not Started"))
+        self.assertEqual(getattr(expected_output[0], "In Progress"), getattr(test_output[0], "In Progress"))
+        self.assertEqual(getattr(expected_output[0], "Complete"), getattr(test_output[0], "Complete"))
+
+    @staticmethod
+    def _generate_2_case_data_rows() -> tuple[Row, Row]:
```
Comment (LJBabbage) [inline test/controllers/test_case_controller.py:59]: This feasibly could be inline, as only used once, same below

**Review by LJBabbage — CHANGES_REQUESTED** [review]
Looks fine, just minor stuff, would like some extra lines in here to make it easier to read. I generally do setup bit (i.e mocks, expected etc), space, command, space, assertions. Just makes it a lot easier to read (at least for me)

File: test/controllers/test_case_controller.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+from test.row_helper import Row
+from unittest import TestCase, mock
+from uuid import UUID
+
+from rm_reporting.controllers import case_controller
+
+exercise_id = "b24a9fd7-0b0d-465f-acf6-cfb8c64e8dfd"
+
+business_1_id = UUID("a9f91050-cd3e-4d29-a14d-67995194fc50")
+business_2_id = UUID("71507170-4090-4f73-bd86-224431e11d3e")
```
[the remark this answers is not in the document]
Comment (insacuri) [reply test/controllers/test_case_controller.py]: Done

File: test/row_helper.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class Row(object):
+    """
+    This is going represent a row returned from SqlAlchemy.  It's not a true representation, but creating a LegacyRow
```
[the remark this answers is not in the document]
Comment (insacuri) [reply test/row_helper.py]: Agreed, updated.

File: test/controllers/test_party_controller.py (hunk 1 above)
Comment (insacuri) [reply test/controllers/test_party_controller.py:101]: Just needed a name and it was the first to come to mind ^^

File: test/controllers/test_case_controller.py (hunk 2 above)
Comment (insacuri) [reply test/controllers/test_case_controller.py:59]: I guess... but I think I want to leave it as is.  I just want to keep the tests themselves easy and short to read.  If we add more tests then we could do something fancier with the test data setup, but for now I think this is fine.

File: test/controllers/test_party_controller.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+survey_id = "0b1e95dd-c84b-4ce3-a6b1-fe6802e4bda0"
+exercise_id = "b24a9fd7-0b0d-465f-acf6-cfb8c64e8dfd"
+
+business_1_id = uuid.UUID("a9f91050-cd3e-4d29-a14d-67995194fc50")
+business_2_id = uuid.UUID("71507170-4090-4f73-bd86-224431e11d3e")
+
+respondent_1_id = uuid.UUID("98bc4edb-c1ef-4335-8d9e-4ef4a7564f7e")
+respondent_2_id = uuid.UUID("877e3b3f-3e5c-4425-b2be-e4cc0b8c2390")
+respondent_3_id = uuid.UUID("f4f18cc6-4b1a-4a88-b6c2-ca3cf38f68fe")
```
[the remark this answers is not in the document]
Comment (insacuri) [reply test/controllers/test_party_controller.py]: Yup 👍

File: test/controllers/test_case_controller.py (hunk 2 above)
Comment (LJBabbage) [reply test/controllers/test_case_controller.py:59]: No problem

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**